### PR TITLE
Add flag to `dune fmt` to disable promoting

### PR DESCRIPTION
**Pull request by gridbugs**
Sometimes you just want to see the diff without applying it.

### CODE REVIEW

**Comment by Alizter** [comment]
Why not just do `dune build @fmt`?

File: bin/build_cmd.ml [hunk 1]
```diff
@@ -202,9 +202,20 @@ let fmt =
     ]
   in
   let term =
-    let+ common = Common.term in
+    let+ common = Common.term
+    and+ no_promote =
+      Arg.(
+        value & flag
+        & info [ "no-promote" ]
```
Comment (rgrinberg) [inline bin/build_cmd.ml:209]: Maybe the flag should be called `preview`?

**Comment by rgrinberg** [comment]
Now that we've added this subcommand, might as well make it complete.

**Comment by gridbugs** [comment]
I did not know about `dune build @fmt`. I imagine many users will follow the same train of thought that led me to write this: noticing that `dune fmt` exists, then noticing that it has side effects, then checking the docs for how to make those side effects not happen.

**Comment by Alizter** [comment]
By the way, there is a `--disable-promotion` flag common to even `dune fmt` that does nothing. I wonder if we should just use that.

**Comment by gridbugs** [comment]
> The problem is that is means something else in `dune build` which is to avoid those auto-promoting rules.

What does it mean to promote a rule?

File: bin/build_cmd.ml (hunk 1 above)
Comment (gridbugs) [reply bin/build_cmd.ml]: yep, changed

**Comment by Alizter** [comment]
@gridbugs I mean that some rules can promote their targets automatically when you build them.

**Comment by gridbugs** [comment]
Makes sense. I think let's name this flag `preview` and keep it separate from `disable-promotion` to avoid confusion (even though in a sense they ideally do the same thing).

**Comment by rgrinberg** [comment]
needs a CHANGES entry

**Comment by gridbugs** [comment]
> needs a CHANGES entry

Added!

**Comment by rgrinberg** [comment]
Could you move the entry to doc/changes and rebase + reformat?

Sorry for the back and forth.

**Comment by gridbugs** [comment]
> Could you move the entry to doc/changes and rebase + reformat?

Done!

**Pull request merged**